algorithm.go

```diff
@@ -72,6 +98,44 @@ func (a Algorithm) String() string {
 	}
 }
 
+func (a Algorithm) IsSupported() bool {
+	return (a >= -39 && a <= -35) || (a >= -8 && a <= -7)
+}
+
+// MarshalCBOR marshals the Algorithm as a CBOR int.
+func (a Algorithm) MarshalCBOR() ([]byte, error) {
+	return encMode.Marshal(int64(a))
+}
```

Comment:
It's not strictly necessary, however it is ther because there is a custom `UnmarshalCBOR`, and it seems prudent to provide this for symmetry, even if it is just a fall-through to `int64` marshalling.